packages/aws-cdk/lib/api/hotswap/common.ts

```diff
@@ -33,6 +34,43 @@ export enum ChangeHotswapImpact {
 
 export type ChangeHotswapResult = HotswapOperation | ChangeHotswapImpact;
 
-export function assetMetadataChanged(change: cfn_diff.ResourceDifference): boolean {
+/**
+ * Represents a change that can be hotswapped.
+ */
+export class HotswappableResourceChange {
+  /**
+   * The value the resource is being updated to.
+   */
+  public readonly newValue: cfn_diff.Resource;
+
+  /**
+   * The changes made to the resource properties.
+   */
+  public readonly propertyUpdates: { [key: string]: cfn_diff.PropertyDifference<any> };
+
+  public constructor(newValue: cfn_diff.Resource, propertyUpdates: { [key: string]: cfn_diff.PropertyDifference<any> }) {
+    this.newValue = newValue;
+    this.propertyUpdates = propertyUpdates;
+  }
+}
+
+export async function establishHotswappableResourceName(
```

Comment:
I don't think "hotswappable" makes too much sense in this name.

Let's call this function `establishResourcePhysicalName()`.